node/src/components/consensus/highway_core/validators.rs

```diff
@@ -126,11 +125,36 @@ impl<VID: Eq + Hash> Validators<VID> {
             |(idx, v): (usize, &'a Validator<VID>)| (ValidatorIndex::from(idx as u32), v.id());
         self.iter().enumerate().map(to_idx)
     }
+
+    pub(crate) fn ensure_nonzero_proposing_stake(&mut self) -> bool {
+        if self.total_weight.is_zero() {
+            return false;
+        }
+        if self.iter().all(|v| v.banned || v.weight.is_zero()) {
```

Comment:
In mainnet it can't happen because there are fewer than 2<sup>64</sup> motes. But if the total stake were more than that, we'd scale down the stakes so they fit in a 64-bit weight. So: Everyone who has less than a 2<sup>64</sup>th of the total stake will end up with weight 0.